### PR TITLE
Add `snap_getEntropy` JSON-RPC method

### PR DESCRIPTION
**Pull request by Mrtenz**
This is an initial implementation of `snap_getEntropy`, as per SIP-6.

### CODE REVIEW

File: packages/snaps-utils/src/manifest/validation.ts [hunk 1]
```diff
@@ -21,6 +21,13 @@ import { CronjobSpecificationArrayStruct } from '../cronjob';
 import { NamespacesStruct } from '../namespace';
 import { NameStruct, NpmSnapFileNames, VersionStruct } from '../types';
 
+// BIP-43 purposes that cannot be used for entropy derivation. These are in the
+// string form, ending with `'`.
+const FORBIDDEN_PURPOSES: string[] = [
+  // SIP-6 magic value.
+  `1399742832'`,
```
Comment (FrederikBolding) [inline packages/snaps-utils/src/manifest/validation.ts:28]: Could this be a shared constant so we don't have to have it defined multiple places?

File: packages/rpc-methods/src/restricted/getBip32Entropy.test.ts [hunk 2]
```diff
@@ -196,6 +197,22 @@ describe('getBip32EntropyCaveatSpecifications', () => {
...
+      await expect(
+        getBip32EntropyCaveatSpecifications[
+          SnapCaveatType.PermittedDerivationPaths
+        ].decorator(fn, {
+          type: SnapCaveatType.PermittedDerivationPaths,
+          value: [params],
+          // @ts-expect-error Missing other required properties.
+        })({ params: { ...params, path: ['m', SIP_6_MAGIC_VALUE, "0'"] } }),
+      ).rejects.toThrow(
+        'Invalid BIP-32 entropy path definition: At path: path -- The purpose "1399742832\'" is not allowed for entropy derivation.',
```
Comment (FrederikBolding) [inline packages/rpc-methods/src/restricted/getBip32Entropy.test.ts:213]: Just to be sure: This uses the snaps-utils validation correct? So no need to change the specification code.

File: packages/rpc-methods/src/restricted/getEntropy.ts [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+/**
+ * Get a BIP-32 derivation path array from a hash, which is compatible with
+ * `@metamask/key-tree`. The hash is assumed to be 32 bytes long.
+ *
+ * @param hash - The hash to derive indices from.
+ * @returns The derived indices as a {@link HardenedBIP32Node} array.
+ */
+function getDerivationPathArray(hash: Uint8Array): HardenedBIP32Node[] {
+  const array: HardenedBIP32Node[] = [];
+  const view = new DataView(hash.buffer, hash.byteOffset, hash.byteLength);
```
Comment (FrederikBolding) [inline packages/rpc-methods/src/restricted/getEntropy.ts:100]: Don't we have a utility for this?

File: packages/rpc-methods/src/restricted/getEntropy.ts [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+
+  for (let index = 0; index < 8; index++) {
+    const uint32 = view.getUint32(index * 4);
+
+    // This is essentially `index | 0x80000000`. Because JavaScript numbers are
+    // signed, we use the bitwise unsigned right shift operator to ensure that
+    // the result is a positive number.
+    // eslint-disable-next-line no-bitwise
+    const pathIndex = (uint32 | HARDENED_VALUE) >>> 0;
+    array.push(`bip32:${pathIndex - HARDENED_VALUE}'` as const);
```
Comment (FrederikBolding) [inline packages/rpc-methods/src/restricted/getEntropy.ts:110]: Are all indices meant to be hardened?

Comment (FrederikBolding) [inline packages/rpc-methods/src/restricted/getEntropy.ts:110]: Are all indices meant to be hardened?

Comment (Mrtenz) [reply packages/rpc-methods/src/restricted/getEntropy.ts:110]: Yes.

Comment (Mrtenz) [reply packages/rpc-methods/src/restricted/getEntropy.ts:110]: Yes.

File: packages/rpc-methods/src/restricted/getBip32Entropy.test.ts (hunk 2 above)
Comment (Mrtenz) [reply packages/rpc-methods/src/restricted/getBip32Entropy.test.ts:213]: Yes.

File: packages/snaps-utils/src/manifest/validation.ts (hunk 1 above)
Comment (Mrtenz) [reply packages/snaps-utils/src/manifest/validation.ts:28]: Yes.

**Review by FrederikBolding — DISMISSED** [review]

**Pull request merged**